### ISSUE TITLE
FR: Move main package from cmd/ into cmd/scip/

### ISSUE THREAD

**Issue by ciarand** [sourcegraph/scip#184]
Hello! This is a feature request to move the main package for the `scip` tool into `cmd/scip` instead of just `cmd`. This is a minor thing, but being able to install this tool via `go install github.com/sourcegraph/scip/cmd/scip@whatever` would be excellent. Doing `go install github.com/sourcegraph/scip/cmd@whatever` works, but it names the binary `cmd` which is awkward and requires renaming.

**Comment by varungandhi-src** [comment]
> `go install github.com/sourcegraph/scip/cmd@whatever` works, but it names the binary cmd which is awkward and requires renaming.

Since the CLI is meant to be used across different ecosystems, we didn't really consider this use case much (and that's why this is not documented in the README)

As such, the supported way of using the tool are:
- If you're purely a downstream user: use one a tagged release archives.
- If you're going to be a contributor: clone the repo and build from source.

Using a random commit may or may not work...

As such, this change would break any existing code relying on `go install github.com/sourcegraph/scip/cmd`. It also makes navigating between files in this repo slightly less convenient, because now both `bindings/go/scip` and `cmd/scip` will be ending in `scip`.

As such, I have a weak -1 for this proposal. I certainly get your point, but I don't want to give the wrong impression that using whatever is on `main` is a supported use case.

@olafurpg and @tjdevries -- do you have any thoughts here? Yay/nay?

**Comment by ciarand** [comment]
>Using a random commit may or may not work...

> I certainly get your point, but I don't want to give the wrong impression that using whatever is on main is a supported use case

I don't think this is specific to a "random commit" or "whatever is on main" and I agree with your points - I wouldn't ask y'all to document that as being a sanctioned installation method for the reasons you laid out above. Right now you can do `go install github.com/sourcegraph/scip/cmd@v0.3.0`, it just spits out a binary named "cmd" instead of "scip". So this is purely about slightly modifying the package structure to make that mechanism work, it's not an endorsement of building off of head. If there were specific build steps that required some extra processing (some sort of generated input that wasn't getting committed to the repo) then that would be a reason to avoid, but in terms of build steps this looks like a pretty vanilla Go project atm.

Not a huge deal ofc, but I figured I'd raise it while y'all are still in the pre-1.0 stage.
